app/models/comment.rb

```diff
@@ -64,6 +75,12 @@ def initialize(person, target, text)
       super(person, target)
     end
 
+    def build(*args)
+      super.tap do |comment|
+        comment.filter_mentions(comment.people_allowed_to_be_mentioned)
```

Comment:
It doesn't seem to be filtered properly at the receive via federation, but that probably also applies to mentions in StatusMessage since the #6902 merge (or earlier). Gonna write some tests for that.